app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ListingsController < ApplicationController
+  before_action :set_listing, only: %i[show edit update destroy]
+
+  def index
+    @listings = Listing.all
+  end
+
+  def new
+    @listing = Listing.new
+  end
+
+  def create
+    @listing = Listing.new(listing_params)
```

Comment:
please add the relation with the user before saving. I would think something like:
`@listing.user = current_user` ?